### PR TITLE
network-reset: Remove note tree size from block header

### PR DESCRIPTION
**Pull request by dguenther**
## Summary

Removes the note tree size from the network-serialized block and PartialBlock, leaving it on the database-stored block and fixtures. We can always do a migration to remove it from the database-stored block if it's no longer necessary, but we use it in some cases now.

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[X] Yes
```

### CODE REVIEW

File: .github/workflows/rust_ci.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
         name: "Clippy check on ironfish-rust"
         with:
           command: clippy
-          args: --manifest-path ironfish-rust/Cargo.toml -- -D warnings
+          args: --manifest-path ironfish-rust/Cargo.toml --all-targets -- -D warnings
```
Comment (danield9tqh) [inline .github/workflows/rust_ci.yml:71]: This belong in this PR?

Comment (dguenther) [reply .github/workflows/rust_ci.yml]: Should be gone now!

File: ironfish/src/blockchain/database/headers.ts [hunk 2]
```diff
@@ -18,8 +18,8 @@ export class HeaderEncoding implements IDatabaseEncoding<HeaderValue> {
 
     bw.writeU32(value.header.sequence)
     bw.writeHash(value.header.previousBlockHash)
-    bw.writeHash(value.header.noteCommitment.commitment)
-    bw.writeU32(value.header.noteCommitment.size)
+    bw.writeHash(value.header.noteCommitment)
+    bw.writeU32(value.header.noteSize ?? 0)
```
Comment (danield9tqh) [inline ironfish/src/blockchain/database/headers.ts:22]: Assert that it is not null here

File: ironfish/src/blockchain/blockchain.ts [hunk 3]
```diff
@@ -1234,7 +1244,7 @@ export class Blockchain {
 
     // If the tree sizes don't match the previous block, we can't verify if the tree
     // sizes on this block are correct
-    const prevNotesSize = prev?.noteCommitment.size || 0
+    const prevNotesSize = prev?.noteSize || 0
```
Comment (danield9tqh) [inline ironfish/src/blockchain/blockchain.ts:1247]: Do we want to assert that `prev?.noteSize` is not null here?

File: ironfish/src/blockchain/database/headers.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ export class HeaderEncoding implements IDatabaseEncoding<HeaderValue> {
     const sequence = reader.readU32()
     const previousBlockHash = reader.readHash()
     const noteCommitment = reader.readHash()
-    const noteCommitmentSize = reader.readU32()
+    const noteSize = reader.readU32() || null
```
[the remark this answers is not in the document]
Comment (dguenther) [reply ironfish/src/blockchain/database/headers.ts]: Done, added an Assert

File: ironfish/src/blockchain/database/headers.ts (hunk 2 above)
Comment (dguenther) [reply ironfish/src/blockchain/database/headers.ts]: Done

**Review by danield9tqh — APPROVED** [review]

**Pull request merged**